open-company/open-company-web
# Deep linking

card: https://trello.com/c/PJxhfRFK

#### This branches from 444-feedbacks and should be merged after #155 

To test section and section editing linking:
- [x] go to /buffer
- [x] go to /buffer/update (if update is a non archived section for buffer)
- [x] go to /buffer/update/edit
- [x] go to /read-only/update/edit while you are logged in: you shouldn't be able to edit
- [x] go to /read-only/update/edit while you are not logged in: you should be redirected to /read-only/update
- [x] go to /buffer and open some sections, navigate using the keyboard arrows btw sections, edit some sections: make sure that the url always reflects the current section and the editing mode

To test history navigation:
- [x] go to /buffer/product
- [x] navigate the history and check that the url is updated to reflect the current as-of
- [x] navigate directly to one revision, ie: /buffer/product?as-of=2015-09-18T18:33:46.000Z
- [x] check that the loaded revision is correct and that you can still navigate the history

To test section navigation inside a stakeholder update:
- [x] create a stakehoder update, let's say the slug is may-2016
- [x] go to /buffer/updates/may-2016
- [x] expand a section, did the url change to /buffer/updates/may-2016/section?
- [ ] use the keyboard arrows to navigate btw sections, did the url change every time the section change?
- [x] close the fullscreen topic, did the url go back to /buffer/updates/may-2016?
- [x] navigate directly to /buffer/updates/may-2016/diversity, did it work?

This also address an issue when loading a Stakeholder Update using an ObscURL, if the user didn't have permission for the company GET it crashed the UI. Now it gets the company data needed to show the SU view from the stakeholder update response.

To test:
- [x] create a stakeholder update for a non public company
- [x] share via URL
- [x] open the URL, can you see it? good!
- [x] open the URL from an incognito window, can you see the SU? good!
- [ ] -(optional) open the URL using a slack user of a different slack org, can you see the SU? good!

Possible problem: right now if you navigate to an old revision the content-type for the get request is hardcoded using this function https://github.com/open-company/open-company-web/pull/156/files#diff-f8be44d03d8bcf180c404c8cac0ad27bR20
We should probably change the server side to accept any kind of content-type to avoid errors in the future if we change the content-type.
